### PR TITLE
chore(design): translate architecture_overview.md in English

### PR DESCRIPTION
**Pull request by takam5f2**

### CODE REVIEW

File: docs/tutorials/create_architecture.md [hunk 1]
```diff
@@ -1,221 +1,223 @@
-# 評価準備
+# Preparation for measuring latency of callback-chains
 
-CARET は、評価する際にアーキテクチャファイルの作成が必要です。
+CARET is capable of measuring latency of not only callback functions and topic communications, but also node chains.
+Specifying a callback function as target is very simple because name is corresponding to the callback function. Topic communication is specified by a topic name, a source node, and a destination node. However, specifying a node chain is more difficult than them. A node chain is defined as a combination of callback functions and topic communications, and often called as "path". Though node chain is regarded as DAG (Direct Acyclic Graph), a node chain may have loops and forks. Loops and forks make a node chain make complicated. Additionally, a large system tends to have many candidates of node chains because it is constructed of many callback functions which are communicated via many topics.
```
Comment (takeshi-iwanari) [inline docs/tutorials/create_architecture.md:4]: Loops and forks make a node chain ~~make~~ complicated.

File: docs/tutorials/create_architecture.md [hunk 2]
```diff
@@ -1,221 +1,223 @@
...
 
-トレース結果からアーキテクチャファイルの雛形を作成できますが、一部手作業での修正が必要になります。  
-ここでは、アーキテクチャファイルの作成方法とその修正方法を説明します。
+## Overview of an architecture file
 
-## アーキテクチャファイルとは
+To deal with such difficulty of specifying node chains, CARET serves configuration for users to choose target node chains. The configuration is defined in a yaml file, which is called "**architecture file**". An architecture file describes a target application's structure. CARET can create an architecture file for minimum description automatically with using measured data, but it does not have target node chains. You will add description for target node chains to a generated architecture file.
```
Comment (takeshi-iwanari) [inline docs/tutorials/create_architecture.md:8]: automatically ~~with~~ using measured data

File: docs/tutorials/create_architecture.md [hunk 3]
```diff
@@ -1,221 +1,223 @@
...
+           subscribe_topic_name: /topic2
+         - node_name: /timer_driven_node
+           publish_topic_name: /topic4
+           subscribe_topic_name: /topic3
+         - node_name: /actuator_dummy_node
+           publish_topic_name: UNDEFINED
+           subscribe_topic_name: /topic4
+   ```
 
 ## ノードレイテンシの定義指定
```
Comment (takeshi-iwanari) [inline docs/tutorials/create_architecture.md:138]: ~~ノードレイテンシの定義指定~~

File: docs/tutorials/create_architecture.md [hunk 4]
```diff
@@ -1,221 +1,223 @@
...
+   '/actuator_dummy_node') # destination node
+   ```
 
-```python
-paths = arch.search_paths(
-    '/sensor_dummy_node', # パスの開始ノード
-    '/actuator_dummy_node') # パスの終了ノード
-```
+   If a target application is large and complicated, `arch.search_paths` method may consume time more than 1 minute.
+   For decreasing consumed time, you can ignore nodes and topics and specify depth of search. Refer to [パスの探索方法](../supplements/how_to_search_path.md) for more details.
```
Comment (takeshi-iwanari) [inline docs/tutorials/create_architecture.md:76]: "パスの探索方法" needs to be translated

File: docs/tutorials/create_architecture.md [hunk 5]
```diff
@@ -1,221 +1,223 @@
...
 
 ## ノードレイテンシの定義指定
 
-ノードレイテンシは、「ノードがメッセージを subscribe し、コールバックが処理開始する時刻」から「ノードがメッセージを publish する時刻」までとしています。  
-ただし、ノードレイテンシはノードの実装にも大きく依存し、統一的な手法での測定は困難です。
+## How to define latency of a single node
+
+Latency of a single node, so called "node latency", is defined as elapsed time between 1. starting time and 2. publishing time as below.
+
+1. starting time when node subscribes topic message and invokes a corresponding callback function
```
Comment (takeshi-iwanari) [inline docs/tutorials/create_architecture.md:144]: starting time -> starting time:

File: docs/tutorials/create_architecture.md (hunk 3 above)
Comment (takam5f2) [reply docs/tutorials/create_architecture.md]: Fixed #36

File: docs/tutorials/create_architecture.md (hunk 1 above)
Comment (takam5f2) [reply docs/tutorials/create_architecture.md]: Fixed in https://github.com/tier4/CARET_doc/pull/36

**Review by hsgwa — CHANGES_REQUESTED** [review]
Added a little more information about TILDE.

File: docs/design/architecture_overview.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+Runtime Data object serves APIs to retrieve `pandas.DataFrame`-based objects including callback latency or communication. Users can analyze temporal aspects of their applications, with visualization, as they expect. APIs for visualization are also served by CARET_analyze which plays main role to analyze trace data.
+
+## Co-design with TILDE, a framework tools to detect deadline overrun
+
+CARET can cooperate with TILDE, a framework tool to detect deadline overrun. As TILDE's tracepoints are embedded in user codes of target applications, TILDE lets CARET trace events in user applications which cannot be traced from ROS/DDS layer. One example of the events in user applications is consumption of topic messages buffered in nodes. CARET can observe subscription and publish of topic messages but not strictly trace consumption of topic messages buffered in nodes. TILDE is able to trace execution of callback function to consume a certain buffered message. With TILDE's capability, CARET can observe such events seen in application layer as well as ROS/DDS layer.
```
[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/design/architecture_overview.md:18]: @hsgwa Thank you for your advice. I'd like to consider to replace a similar sentence by suggested one.

[the remark this answers is not in the document]
Comment (takam5f2) [reply docs/design/architecture_overview.md]: @hsgwa fixed in https://github.com/tier4/CARET_doc/pull/38/commits/0b5596cba229446e58386daf778293b840d7a684
@y-okumura-isp Could you check contents of "cooperation with TILDE" if you can ? Thanks,

**Review by hsgwa — APPROVED** [review]
LGTM

**Comment by takam5f2** [comment]
@hsgwa  
Sorry, could you approve this PR. I resolved conflict after this PR is approved.

**Review by hsgwa — APPROVED** [review]

**Pull request merged**